src/vsphere_cpi/lib/cloud/vsphere/resources/cluster.rb

```diff
@@ -81,40 +81,22 @@ def select_datastores(pattern)
 
       # Fetches the raw cluster utilization from vSphere.
       #
-      # First filter out any hosts that are in maintenance mode. Then aggregate
-      # individual host capacity and its utilization using the performance
-      # manager.
-      #
-      # @param [Array<Vim::HostSystem>] cluster_host_systems cluster hosts.
-      # @return [void]
-      def fetch_cluster_utilization(cluster_host_systems)
-        hosts_properties = @client.cloud_searcher.get_properties(
-          cluster_host_systems, Vim::HostSystem, HOST_PROPERTIES, ensure_all: true)
-        active_host_mobs = select_active_host_mobs(hosts_properties)
-
-        synced_free_memory = 0
-        return synced_free_memory if active_host_mobs.empty?
-
-        cluster_free_memory = 0
-
-        counters = @client.get_perf_counters(active_host_mobs, HOST_COUNTERS, max_sample: 5)
-        counters.each do |host_mob, counter|
-          host_properties = hosts_properties[host_mob]
-          total_memory = host_properties['hardware.memorySize'].to_i
-          free_memory = 0
-          if !counter['mem.usage.average'].nil?
-            percent_used = Util.average_csv(counter['mem.usage.average']) / 10000
-            free_memory = ((1.0 - percent_used) * total_memory).to_i
-          else
-            logger.warn("host '#{host_properties['name']}' is missing 'mem.usage.average' (possibly recently booted), ignoring")
-          end
-
-          cluster_free_memory += free_memory
+      # @return
+      def fetch_cluster_utilization()
```

Comment:
Method name should be effective_memory as thats what this is returning

Reply:
This was earlier also named as cluster utilization. I would prefer to keep changes as minimal as possible. If we ever introduce cluster CPU usage as one of the requirements , then I can rename things differently.
What do you think?